### PR TITLE
Renaming project name in compiling guides

### PR DESCRIPTION
**Pull request by Firestorm01X2**
Renaming project name in compiling guides

#### Summary
SUMMARY: Build "Renaming project name in compiling guides"

#### Purpose of change
There are path to CDDA github all over the place. Also project name referenced as Cataclysm-DDA.

#### Describe the solution
Change project name from Cataclysm-DDA to Cataclysm-BN.
Change path from CDDA repo to BN repo.
Other misc changes.

#### Describe alternatives you've considered
None.

#### Testing
Observe the documentation

#### Additional context
 doc/COMPILING/COMPILING-FLATPAK.md is not working. Added text about it.
I've intentonally pass few things, that I've not sure. For example directory paths - it is should be examined separately.

### CODE REVIEW

File: doc/COMPILING/COMPILING.md [hunk 1]
```diff
@@ -494,7 +494,7 @@ The Cataclysm source is compiled using `make`.
 * `RELEASE=1` build an optimized release version; omit for debug build.
 * `CLANG=1` build with [Clang](http://clang.llvm.org/), the compiler that's included with the latest Command Line Tools for Xcode; omit to build using gcc/g++.
 * `MACPORTS=1` build against dependencies installed via Macports, currently only `gettext` and `ncurses`.
-* `USE_HOME_DIR=1` places user files (config, saves, graveyard, etc) in the user's home directory. For curses builds, this is `/Users/<user>/.cataclysm-dda`, for SDL builds it is `/Users/<user>/Library/Application Support/Cataclysm`.
+* `USE_HOME_DIR=1` places user files (config, saves, graveyard, etc) in the user's home directory. For curses builds, this is `/Users/<user>/.cataclysm-bn`, for SDL builds it is `/Users/<user>/Library/Application Support/Cataclysm`.
```
Comment (Firestorm01X2) [inline doc/COMPILING/COMPILING.md:497]: I hope it is that way. It should be.

File: doc/COMPILING/COMPILING-FLATPAK.md [hunk 2]
```diff
@@ -1,8 +1,9 @@
-The supplied flatpak manifest org.cataclysmdda.CataclysmDDA.json will
-download and build the latest release of Cataclysm-DDA.  An app built in
+** This way is currently unavalible **
```
Comment (Firestorm01X2) [inline doc/COMPILING/COMPILING-FLATPAK.md:1]: Because even old link is not working. Looks like this way of compiling heavily outdated.

**Comment by Firestorm01X2** [comment]
Well, it it ready. Just look for my comments - something could be changed.

File: doc/COMPILING/COMPILING-FLATPAK.md (hunk 2 above)
Comment (Coolthulhu) [reply doc/COMPILING/COMPILING-FLATPAK.md:1]: This section may need a purge in the future. But for now, the note is better than nothing.
Maybe I could try fixing that flatpack thing one day, but it's very distant future at best.

**Pull request merged**